src/app/settings/actions.ts

```diff
@@ -1,14 +1,20 @@
-import { createAction } from 'typesafe-actions';
+import { createAction, PayloadAction } from 'typesafe-actions';
 import { Settings } from './reducer';
 
 /** Bulk update settings after they've been loaded. */
 export const loaded = createAction('settings/LOADED')<Partial<Settings>>();
 
 /** This one seems a bit like cheating, but it lets us set a specific property. */
-export const setSetting = createAction('settings/SET', (property: keyof Settings, value: any) => ({
-  property,
-  value
-}))();
+export const setSetting = createAction(
+  'settings/SET',
+  <V extends keyof Settings>(property: V, value: Settings[V]) => ({
+    property,
+    value
+  })
+)() as <V extends keyof Settings>(
```

Comment:
This is a bit of a brain bender but it allows us to type `value` based on the key provided in `property` which is pretty awesome.